sdks/go/pkg/beam/core/timers/timers_test.go

```diff
@@ -0,0 +1,189 @@
+// Licensed to the Apache SoFiringTimestampware Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, soFiringTimestampware
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package timers
+
+import (
+	"testing"
+	"time"
+
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/core/graph/mtime"
+	"github.com/google/go-cmp/cmp"
+)
+
+type fakeTimerProvider struct {
+	timerDetails TimerMap
+}
+
+func (t *fakeTimerProvider) Set(tm TimerMap) {
+	t.timerDetails = tm
+}
+
+func (t *fakeTimerProvider) GetMap() TimerMap {
+	return t.timerDetails
+}
+
+func TestProcessingTimeTimer_Set(t *testing.T) {
+	type fields struct {
+		Key  string
+		Kind timeDomainEnum
+	}
+	type args struct {
+		FiringTimestamp mtime.Time
+		opts            []timerOptions
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		out    TimerMap
+	}{
+		{
+			name:   "correct set no options",
+			fields: fields{Key: "Window", Kind: TimeDomainProcessingTime},
+			args: args{
+				FiringTimestamp: mtime.FromTime(time.Now()),
+			},
+			out: TimerMap{
+				Key:           "Window",
+				FireTimestamp: mtime.FromTime(time.Now()),
+			},
+		},
+		{
+			name:   "correct set tag option",
+			fields: fields{Key: "Window", Kind: TimeDomainProcessingTime},
+			args: args{
+				FiringTimestamp: mtime.FromTime(time.Now()),
+				opts:            []timerOptions{WithTag("windowTag")},
+			},
+			out: TimerMap{
+				Key:           "Window",
+				FireTimestamp: mtime.FromTime(time.Now()),
+				Tag:           "windowTag",
+			},
+		},
+		{
+			name:   "correct set tag and timestamp option",
+			fields: fields{Key: "Window", Kind: TimeDomainProcessingTime},
+			args: args{
+				FiringTimestamp: mtime.FromTime(time.Now()),
+				opts:            []timerOptions{WithTag("windowTag"), WithOutputTimestamp(mtime.FromTime(time.Now().Add(2 * time.Second)))},
+			},
+			out: TimerMap{
+				Key:           "Window",
+				FireTimestamp: mtime.FromTime(time.Now()),
+				Tag:           "windowTag",
+				HoldTimestamp: mtime.FromTime(time.Now().Add(2 * time.Second)),
```

Comment:
Rather than setting time.Now().Add(2 * time.Second)) twice, we should probably store that as a value before entering the test. Otherwise, there could be some drift between the 2 invocations

Reply:
Same comment applies in the other test